ct-app/core/components/hoprd_api.py

```
@@ -1,256 +1,178 @@
 
         return (False, None)
 
     async def __call_api(
-        self,
-        obj: Callable[..., object],
-        method: str,
-        call_log: bool = True,
-        timeout: int = 60,
-        *args,
-        **kwargs,
+        self, method: Method, endpoint: str, data: dict = {}, timeout: int = 60
```

Ruff

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)